### ISSUE TITLE
New name proposal: Ruins (RUst INteractive Shell)

### ISSUE THREAD

**Issue by digital-diplomat** [evcxr/evcxr#310]
The difficulty I have remembering the exact name of `evcxr` to run it as an eval context is hard to overstate. The current name is not intuitive and, though it is meant to be mnemonic (*Ev*al *C*onte*x*t for *R*ust), it reads to me as a random mix of letters whose order I can never remember.

I'm proposing the name **Ruins** (*Ru*st *In*teractive *S*hell) as an alternative that is a real dictionary word, unused by any other development software that I know of. It elicits a similar mental image to the name "Rust" (old weathered things) which further solidifies the connection between the two.

`ruins` is what I currently have aliased to pull up `evcxr` in my shell when I want to mess around with Rust code, and a solid name and identity can encourage further user adoption and support.

**Comment by davidlattimore** [comment]
I agree that the current name definitely isn't ideal. While I wouldn't rule out changing the name, I feel like changing the name wouldn't be something to take lightly. There's 5 years of history of use of the current name. People have written blog posts, a journal article and a book that references the current name. I'm also not sure how the upgrade-path would look like for people who already have evcxr installed.

Ruins seems like a reasonable name to me, but there's already a crate named `ruins` on crates.io.

I feel like if we were to rename it, I'd want to perhaps merge the repl and jupyter kernel into a single binary at the same time.

I'm unsure how much having a different name would help with further user adoption. I doubt there are many people who would choose not to use a tool just because of its name - especially given that it's relatively easy to just create an alias or symlink. I think discoverability is probably a bigger issue - and discoverability is, I think, mostly orthogonal to the name.

**Comment by digital-diplomat** [comment]
I definitely understand that it's not something that can just *happen*, nor should it. I wanted to open this issue as a means of starting that conversation.

This program is basically the *de facto* standard REPL/eval context for Rust, so having it be easily accessible and memorable, especially to those who are newly learning the language, can be extremely valuable.

The crate already in place is currently just a "Hello World" project that has not been updated or touched for over a year. If it's not possible to replace the program, then maybe keeping the package name as `evcxr` but calling the executable `ruins` could be the appropriate move.

I'm unsure what challenges a merge of the REPL and Jupyter kernel could create, but I don't doubt having one solid "download *this* package" package would be a very good idea.

And to cover that last point, you're absolutely right that a simple name change isn't going to make or break user adoption…but it's a nice little addition to the user experience that I think many users would at least find pleasant. :)

**Comment by davidlattimore** [comment]
I was thinking about this some more and realised that if I created a new combined REPL and Jupyter kernel as a single binary with a new name, I could leave evcxr_repl and evcxr_jupyter as standalone versions that would continue to be updated. Then, if at some point it becomes apparent that most people have switched to the new combined binary, the old standalone binaries could be deprecated, but if that didn't happen, they could continue to be updated indefinitely.

As for what the new name should be, I'm unsure. `ruins` is a possibility, but I'd want to get wider input from the community as to what the new name should be. I think whatever it's called, I'd want the binary name to be the same as the crate name, but you're right that it does look like `ruins` is probably not being actively developed, so the author may be open to transferring the crate name.

Also, note that there's issue #23 that discusses the naming issue. I didn't do anything back then because evcxr was still under the google github org at that point and it just seemed too hard.

**Comment by digital-diplomat** [comment]
Looking at that older issue, I also *really* like `rusti` as a potential name. It's kind of cute and approachable, but not unprofessionally so.

I *absolutely* think community input is warranted for this kind of change. Worst case scenario, if we did go with `ruins` but the crate name can't be made available, we could always name the crate `ruins_eval` or something similar so that the binary and crate aren't *entirely* different, similar to how `evcxr` is a binary in the `evcxr_repl` crate.

And since I'm covering your points in reverse order for some reason, I really like the idea of having the name change coincide with a combination of the two binaries. It makes intuitive sense that if we're trying to simultaneously rebrand and restructure the program(s), we should do so in one fell swoop.

**Comment by davidlattimore** [comment]
> Looking at that older issue, I also _really_ like `rusti` as a potential name. It's kind of cute and approachable, but not unprofessionally so.

The ship sailed on that one - there's been another REPL with that name for some time now.

**Comment by elenakrittik** [comment]
I personally don't feel like the current name is problematic, but wanted to say that it's fairly easy to alias the `evcxr` binary to `ruins` if you decide to do it. This will only require specifying two `[[bin]]` sections with different `name = "..."`s for the same `path = "<main.rs path>"`.
